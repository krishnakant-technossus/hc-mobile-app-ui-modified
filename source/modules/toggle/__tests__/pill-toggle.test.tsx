// npx jest ./source/modules/toggle/__tests__/pill-toggle.test.ts

import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import PillToggle from '../pill-toggle';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

describe('PillToggle', () => {
  let onPressMock: any, options: any;

  beforeEach(() => {
    onPressMock = jest.fn();

    options = [
      {label: 'Selected', value: 1, onPress: onPressMock},
      {label: 'Unselected', value: 2, onPress: onPressMock},
      {label: 'Disabled', value: 3, onPress: onPressMock, disabled: true},
    ];
  });

  describe('Rendering', () => {
    it('renders itself correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <PillToggle selected={1} options={options} />
        </ThemeProvider>,
      );

      const pillToggle = wrapper.getByTestId('pill-toggle');

      expect(pillToggle.children).toHaveLength(options.length);

      expect(pillToggle).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });

    it('renders the children correctly', () => {
      const {getByText} = render(
        <ThemeProvider theme={defaultTheme}>
          <PillToggle selected={1} options={options} />
        </ThemeProvider>,
      );

      const option1 = getByText(options[0].label);
      const option2 = getByText(options[1].label);
      const option3 = getByText(options[2].label);

      expect(option1).toBeDefined();
      expect(option2).toBeDefined();
      expect(option3).toBeDefined();
    });
  });

  describe('Interactions', () => {
    it('should call onPress function on click of pill button', () => {
      const {getAllByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <PillToggle selected={1} options={options} />
        </ThemeProvider>,
      );

      const buttons = getAllByTestId('pill-toggle-button');

      fireEvent.press(buttons[0]);

      expect(onPressMock).toHaveBeenCalledTimes(1);
    });

    it('should not call onPress function on click of pill button when option is disabled', () => {
      const {getAllByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <PillToggle selected={1} options={options} />
        </ThemeProvider>,
      );

      const buttons = getAllByTestId('pill-toggle-button');

      fireEvent.press(buttons[2]);

      expect(onPressMock).not.toHaveBeenCalled();
    });

    it('should not call onPress function on click of pill button when toggle itself is disabled', () => {
      const {getAllByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <PillToggle selected={1} options={options} disabled />
        </ThemeProvider>,
      );

      const buttons = getAllByTestId('pill-toggle-button');

      fireEvent.press(buttons[0]);
      fireEvent.press(buttons[1]);
      fireEvent.press(buttons[2]);

      expect(onPressMock).not.toHaveBeenCalled();
    });
  });
});
