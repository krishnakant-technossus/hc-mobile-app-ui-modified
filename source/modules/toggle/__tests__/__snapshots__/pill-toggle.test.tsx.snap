// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PillToggle Rendering renders itself correctly 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
      },
    ]
  }
  testID="pill-toggle"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#00AEFF",
        "borderBottomLeftRadius": 22,
        "borderBottomRightRadius": 0,
        "borderColor": "#00AEFF",
        "borderRightWidth": 0,
        "borderStyle": "solid",
        "borderTopLeftRadius": 22,
        "borderTopRightRadius": 0,
        "borderWidth": 1,
        "opacity": 1,
      }
    }
    testID="pill-toggle-button"
  >
    <Text
      inversed={true}
      muted={false}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "paddingHorizontal": 22,
            "paddingVertical": 10,
            "textAlign": "left",
          },
        ]
      }
      testID="pill-toggle-button-label"
    >
      Selected
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 0,
        "borderBottomRightRadius": 0,
        "borderColor": "#DBE0E3",
        "borderRightWidth": 0,
        "borderStyle": "solid",
        "borderTopLeftRadius": 0,
        "borderTopRightRadius": 0,
        "borderWidth": 1,
        "opacity": 1,
      }
    }
    testID="pill-toggle-button"
  >
    <Text
      inversed={false}
      muted={true}
      style={
        Array [
          Object {
            "color": "#97A4AB",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "paddingHorizontal": 22,
            "paddingVertical": 10,
            "textAlign": "left",
          },
        ]
      }
      testID="pill-toggle-button-label"
    >
      Unselected
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#EDF1F5",
        "borderBottomLeftRadius": 0,
        "borderBottomRightRadius": 22,
        "borderColor": "#DBE0E3",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopLeftRadius": 0,
        "borderTopRightRadius": 22,
        "borderWidth": 1,
        "opacity": 1,
      }
    }
    testID="pill-toggle-button"
  >
    <Text
      inversed={false}
      muted={true}
      style={
        Array [
          Object {
            "color": "#97A4AB",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "paddingHorizontal": 22,
            "paddingVertical": 10,
            "textAlign": "left",
          },
        ]
      }
      testID="pill-toggle-button-label"
    >
      Disabled
    </Text>
  </View>
</View>
`;
